ggu/main.c

```diff
@@ -151,20 +143,44 @@ process_single_packet(struct rte_mbuf *pkt, const struct ggu_config *ggu_conf)
 			goto free_packet;
 		}
 
-		udphdr = (struct udp_hdr *)&ip4hdr[1];
+		minimum_size += ipv4_hdr_len(ip4hdr) - sizeof(*ip4hdr);
+		if (pkt->data_len < minimum_size) {
+			RTE_LOG(NOTICE, GATEKEEPER,
+				"ggu: the IPv4 packet's actual size is %hu, which doesn't have the minimum expected size %hu\n",
+				pkt->data_len, minimum_size);
+			goto free_packet;
+		}
+
+		/*
+		 * The ntuple filter supports for IPv4 variable headers.
```

Comment:
Drop the "for"; final sentence: `The ntuple filter supports IPv4 variable headers.`

Do the same in IPv6 case.